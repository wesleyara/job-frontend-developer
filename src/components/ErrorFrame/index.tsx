import { MdErrorOutline } from "react-icons/md";

interface ErrorFrameProps {
  error: Error | null;
}

export const ErrorFrame = ({ error }: ErrorFrameProps) => {
  return (
    <section
      data-cy="error-frame"
      className="mt-5 w-full px-3 flex flex-col items-center gap-3"
    >
      <MdErrorOutline size={50} />
      <span className="text-center">{error?.message}</span>
    </section>
  );
};
