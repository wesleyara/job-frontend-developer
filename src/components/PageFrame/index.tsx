export const PageFrame = () => {
  return (
    <main className="h-[100vh] flex flex-col items-center justify-center bg-[#ebebeb]"></main>
  );
};
