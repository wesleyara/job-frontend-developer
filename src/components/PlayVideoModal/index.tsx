import { Dialog, DialogPanel, DialogTitle } from "@headlessui/react";
import { YoutubeDetailsType } from "~/types";
import { Dispatch, SetStateAction } from "react";

interface PlayVideoModalProps {
  item: YoutubeDetailsType;
  isOpen: boolean;
  setIsOpen: Dispatch<SetStateAction<boolean>>;
}

export const PlayVideoModal = ({
  item,
  isOpen,
  setIsOpen,
}: PlayVideoModalProps) => {
  return (
    <Dialog
      open={isOpen}
      onClose={() => setIsOpen(false)}
      className="relative z-50"
    >
      <div className="fixed inset-0 flex mx-auto items-center justify-center p-4">
        <DialogPanel className="space-y-4 border border-gray-300 bg-white p-12 max-w-[768px] w-[90%] text-center rounded-md relative">
          <button
            data-cy="close-youtube-modal"
            className="absolute top-2 right-3"
            onClick={() => setIsOpen(false)}
          >
            X
          </button>
          <DialogTitle className="text-lg">{item.snippet.title}</DialogTitle>

          <iframe
            data-cy="youtube-player"
            src={`https://www.youtube.com/embed/${item.id}`}
            title={item.snippet.title}
            className="inset-0 w-full h-[200px] sm:h-[400px]"
            allowFullScreen
          />
        </DialogPanel>
      </div>
    </Dialog>
  );
};
