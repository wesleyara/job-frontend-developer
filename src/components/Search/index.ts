import { SearchButton } from "./SearchButton";
import { SearchInput } from "./SearchInput";
import { SearchRoot } from "./SearchRoot";

export const Search = {
  Root: SearchRoot,
  Button: SearchButton,
  Input: SearchInput,
};
