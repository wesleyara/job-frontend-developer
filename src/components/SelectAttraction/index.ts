import { SelectAttractionButton } from "./SelectAttractionButton";
import { SelectAttractionContent } from "./SelectAttractionContent";
import { SelectAttractionRoot } from "./SelectAttractionRoot";

export const SelectAttraction = {
  Root: SelectAttractionRoot,
  Content: SelectAttractionContent,
  Button: SelectAttractionButton,
};
