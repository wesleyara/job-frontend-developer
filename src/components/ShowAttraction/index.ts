import { ShowAttractionBadges } from "./ShowAttractionBadges";
import { ShowAttractionInfo } from "./ShowAttractionInfo";
import { ShowAttractionRoot } from "./ShowAttractionRoot";

export const ShowAttraction = {
  Root: ShowAttractionRoot,
  Info: ShowAttractionInfo,
  Badges: ShowAttractionBadges,
};
