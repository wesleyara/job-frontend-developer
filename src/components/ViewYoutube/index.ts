import { ViewYoutubeCard } from "./ViewYoutubeCard";
import { ViewYoutubeRoot } from "./ViewYoutubeRoot";

export const ViewYoutube = {
  Root: ViewYoutubeRoot,
  Card: ViewYoutubeCard,
};
