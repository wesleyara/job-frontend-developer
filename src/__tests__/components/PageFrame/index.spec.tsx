import { render } from "@testing-library/react";
import { PageFrame } from "~/components/PageFrame";
import { beforeEach, describe, expect, it } from "vitest";

describe("PageFrame", () => {
  let sut: JSX.Element;

  beforeEach(() => {
    sut = <PageFrame />;
  });

  it("should be defined", () => {
    expect(sut).toBeDefined();
  });

  it("should have a main element", () => {
    const { container } = render(sut);

    expect(container.querySelector("main")).not.toBeNull();
  });

  it("should render correctly", () => {
    const { container } = render(sut);

    expect(container).toMatchSnapshot();
  });
});
