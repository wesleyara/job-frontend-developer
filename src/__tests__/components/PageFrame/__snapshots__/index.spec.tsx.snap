// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageFrame > should render correctly 1`] = `
<div>
  <main
    class="h-[100vh] flex flex-col items-center justify-center bg-[#ebebeb]"
  />
</div>
`;
